client/pages/home.jsx

```diff
@@ -43,41 +45,159 @@ const data = [
 
 export default class Home extends React.Component {
 
+  constructor(props) {
+    super(props);
+    this.state = {
+      error: '',
+      weight: '',
+      editing: false,
+      dayId: '',
+      loading: true
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleClick = this.handleClick.bind(this);
+    this.loadData = this.loadData.bind(this);
+  }
+
   componentDidMount() {
-    // Fetch user data here
+    this.loadData();
+    this.setState({
+      loading: false
+    });
```

Comment:
`this.loadData` wont finish before the setState finishes, you should set loading to `false` once you get your data below.